src/panels/lovelace/cards/hui-clock-card.ts

```diff
@@ -0,0 +1,167 @@
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { DateTime } from "luxon";
+import "../../../components/ha-alert";
+import "../../../components/ha-card";
+import type { HomeAssistant } from "../../../types";
+import type { LovelaceCard, LovelaceCardEditor } from "../types";
+import type { ClockCardConfig } from "./types";
+
+@customElement("hui-clock-card")
+export class HuiClockCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import("../editor/config-elements/hui-clock-card-editor");
+    return document.createElement("hui-clock-card-editor");
+  }
+
+  public static getStubConfig(): ClockCardConfig {
+    return {
+      type: "clock",
+      time_format: "hh:mm:ss",
+      clock_size: "medium",
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property({ type: Boolean }) public preview = false;
+
+  @state() private _config?: ClockCardConfig;
+
+  @state() private _time?: DateTime;
+
+  @state() private _interval = 1000;
+
+  private _tickInterval?: number;
+
+  public setConfig(config: ClockCardConfig): void {
+    if (!config.time_format) {
+      throw new Error("time_format required");
+    }
+
+    this._config = config;
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._startTick();
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._stopTick();
+  }
+
+  private _startTick() {
+    this._tick();
+    this._tickInterval = window.setInterval(() => this._tick(), this._interval);
+  }
+
+  private _stopTick() {
+    if (this._tickInterval) {
+      clearInterval(this._tickInterval);
+      this._tickInterval = undefined;
+    }
+  }
+
+  private _tick() {
+    const locale = this.hass?.locale?.language;
+    const timeZone = this.hass?.config.time_zone;
+
+    this._time = DateTime.now();
+
+    if (locale) this._time = this._time.setLocale(locale);
+    if (timeZone) this._time = this._time.setZone(timeZone);
```

Comment:
```suggestion
    let time = DateTime.now();

    if (locale) time = time.setLocale(locale);
    if (timeZone) time = time.setZone(timeZone);
    this._time = time;
```
We prefer to set the state only one time.